src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorDiagnosticsPublisher.cs

```diff
@@ -281,11 +279,13 @@ private void PublishDiagnosticsForFilePath(string filePath, IEnumerable<Diagnost
                 Host = string.Empty,
             };
 
-            _languageServer.PublishDiagnostics(new PublishDiagnosticsParams()
-            {
-                Uri = uriBuilder.Uri,
-                Diagnostics = new Container<Diagnostic>(diagnostics),
-            });
+            _ = _languageServer.SendNotificationAsync(
+                "textDocument/publishDiagnostics",
```

Comment:
If this is using `Microsoft.VisualStudio.LanguageServer.Protocol`, you can replace this with one of the constants defined there:

```suggestion
                Methods.TextDocumentPublishDiagnosticsName,
```